### PR TITLE
Integration test with sample kbc

### PR DESCRIPTION
**Pull request by Xynnn007**
Due to https://github.com/confidential-containers/image-rs/issues/43
New features:
* Add integration tests for simple signing using sample-kbc
* Refactor integration test of pulling an encrypted image
* Documentation

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+## Image Decryption
+
+Implemented in `decrypt_layer.rs`.
+
+Image decryption will involve the image `"docker.io/arronwang/busybox_kbs_encrypted"`,
+following the steps:
+
+* Pull manifefst of `"docker.io/arronwang/busybox_kbs_encrypted"` without verification of signature.
+* Pull layers of the mentioned image.
+* Ocicrypt-rs will ask the Attestation-Agent to decrypt the LEK, which is
```
Comment (jialez0) [inline tests/README.md:16]: KEK and Lek are not standard terms, so it is best to explain before using here.

File: tests/signature_verification.rs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+/// Name of different signing schemes.
+const SIMPLE_SIGNING: &str = "Simple Signing";
+const NONE_SIGNING: &str = "None";
+
+/// Script for preparing Simple Signing GPG signature file.
+const SIGNATURE_SCRIPT: &str = "scripts/install_test_signatures.sh";
+
+/// Decrypt config
+const DECRYPT_CONFIG: &str = "provider:attestation-agent:sample_kbc::null";
```
Comment (jialez0) [inline tests/signature_verification.rs:23]: The name `DECRYPT_CONFIG` is not clear here. Maybe you can use AA_PARAMETERS or other.

**Review by jialez0 — APPROVED** [review]

File: tests/README.md (hunk 1 above)
Comment (Xynnn007) [reply tests/README.md:16]: Sure!

File: tests/signature_verification.rs (hunk 2 above)
Comment (Xynnn007) [reply tests/signature_verification.rs]: Sure, it will seem clearer

**Comment by Xynnn007** [comment]
Update AA's `get_resource` port as default value.

File: tests/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# Integration Test for Image-rs
+
+This integration test has 2 main sub-type tests:
+* Image decryption using [ocicrypt-rs](https://github.com/confidential-containers/ocicrypt-rs)
+* Image signature verification.
+
+## Image Decryption
+
+Implemented in `decrypt_layer.rs`.
```
Comment (arronwy) [inline tests/README.md:9]: Maybe rename `decrypt_layers.rs` to `decrypt_kbs_encrypted_image.rs` or `image_decryption.rs` will be better?

File: tests/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+Image decryption will involve the image `"docker.io/arronwang/busybox_kbs_encrypted"`,
+following the steps:
+
+* Pull manifefst of `"docker.io/arronwang/busybox_kbs_encrypted"` without verification of signature.
```
Comment (arronwy) [inline tests/README.md:14]: We can also link https://github.com/confidential-containers/image-rs/blob/main/test_data/generate_test_data.md here for how to generate KBS encrypted image

**Review by arronwy — APPROVED** [review]
Thanks @Xynnn007 LGTM, only some naming suggestions.

File: tests/README.md (hunk 3 above)
Comment (Xynnn007) [reply tests/README.md:9]: Yes, changed!

**Pull request merged**